src/plugin/class-subject.php

```diff
@@ -72,5 +72,13 @@ public function store_reference( int $transformed_post_id, string $unique_plugin
 		// Store a reference to your transformation in the source
 		$meta_key = Transformer::META_KEY_LIBERATED_OUTPUT . '_' . $unique_plugin_slug;
 		update_post_meta( $this->id, $meta_key, $transformed_post_id );
+
+		// Control preview
+		add_filter(
+			'data_liberation_preview_transformed_post_id',
+			function () use ( $transformed_post_id ) {
+				return $transformed_post_id;
```

Comment:
I'm having some trouble understanding the purpose of this filter. What would be a use case where something other than `$transformed_post_id` would be returned?

Reply:
`$transformed_post_id` is the native transformed post id with filter on top of it, so plugins can modify its value to have their transformed post id be returned and hence used for preview.

Reply:
The result is obviously ambiguous when multiple plugins are competing since only value can be returned and shown in preview. The winning one would be the one that hooks last & would overwrite the value.

Reply:
Could you give an example of a use case where the plugin would need to do both the following:

1. Call `store_reference()` with a `$transformed_post_id` with value A
2. Return a different value than `A` in the `data_liberation_preview_transformed_post_id` filter

In what situations do you see this happening?

Reply:
Literally every case :) Lets say X plugin takes over S subject type and run its transformation logic to produce P post_id post. We have also produced N post_id post with our native transformation.

With current approach, P doesn't automatically gets used for preview. So, this filter comes into play, where the plugin author would first plug in their own transformation and then return P as the post_id using the filter so that gets used for preview. Hope that makes it clear?

Reply:
Thank you for clarifying. If I'm understanding this correctly, that means that the decision of which transformed post (N or P) is used for the preview becomes the responsibility of plugins?

If so, what advantages does that bring compared to it being the responsibility of "core try-wordpress" to decide what gets used as the preview?

In other words, why should it be a plugin that decides what transformed post gets used for preview? To me this feels like a core responsibility of try-wordpress, I'm not sure I see the advantages of exposing that concept to plugins.